src/pages/project/project_wizard/ProjectFundingsField.tsx

```diff
@@ -0,0 +1,133 @@
+import CancelIcon from '@mui/icons-material/Cancel';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
+import { Box, Button, IconButton, TextField } from '@mui/material';
+import { Field, FieldArray, FieldArrayRenderProps, FieldProps } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { VerifiedFundingApiPath } from '../../../api/apiPaths';
+import { FundingSourceField } from '../../../components/FundingSourceField';
+import { NfrProjectSearch } from '../../../components/NfrProjectSearch';
+import { emptyProjectFunding, ProjectFieldName, ProjectFunding } from '../../../types/project.types';
+import { SpecificFundingFieldNames } from '../../../types/publicationFieldNames';
+import { API_URL } from '../../../utils/constants';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { fundingSourceIsNfr, getNfrProjectUrl } from '../../registration/description_tab/projects_field/projectHelpers';
+
+interface FundingsFieldProps {
+  currentFundings: ProjectFunding[];
+}
+
+export const ProjectFundingsField = ({ currentFundings }: FundingsFieldProps) => {
+  const { t } = useTranslation();
+
+  return (
+    <Box sx={{ marginTop: '1rem' }}>
+      <FieldArray name={ProjectFieldName.Funding}>
+        {({ name, remove, push, form: { setFieldValue } }: FieldArrayRenderProps) => (
+          <Box sx={{ display: 'flex', flexDirection: 'column', gap: '0.5rem' }}>
+            {currentFundings.map((funding, index) => {
+              const baseFieldName = `${name}[${index}]`;
+              const hasSelectedSource = !!funding.source;
+              const hasSelectedNfrSource = fundingSourceIsNfr(funding.source);
+              const fundingId =
+                hasSelectedNfrSource && funding.identifier
+                  ? `${API_URL}${VerifiedFundingApiPath.Nfr.substring(1)}/${funding.identifier}` // TODO: Remove this when(/if) NP-43030 is solved
+                  : '';
+
+              const hasSelectedNfrProject = hasSelectedNfrSource && fundingId;
+
+              return (
+                <Box
+                  key={index}
+                  sx={{
+                    display: 'grid',
+                    gridTemplateColumns: { xs: '1fr', md: '4fr 5fr 2fr 1fr' },
+                    gap: '1rem',
+                    alignItems: 'center',
+                  }}>
+                  <FundingSourceField fieldName={`${baseFieldName}.${SpecificFundingFieldNames.Source}`} />
+
+                  {hasSelectedNfrSource &&
+                    (hasSelectedNfrProject ? (
+                      <>
+                        <TextField
+                          value={funding.identifier}
+                          disabled={hasSelectedNfrSource}
+                          label={t('common.id')}
+                          fullWidth
+                          variant="filled"
+                          data-testid={dataTestId.registrationWizard.description.fundingIdField}
```

Comment:
Alle data-testid skulle vært mot `projectWizard` nå?
```suggestion
                          data-testid={dataTestId.projectWizard.fundingIdField}
```

Reply:
Jeg er usikker, jeg har latt dem være med vilje fordi du en gang før snakket om at vi kunne bruke de samme test-id'ene flere plasser? de er ofte ganske generelle i navn (ala create-user-button), så kanskje det ikke er noen grunn til å ha flere like med ulike stier? Men samtidig er jeg enig i at det er en uting med spor etter hvor det er kopiert fra da.

Reply:
Sant det ja. Hadde jeg startet på nytt nå tror jeg både `dataTestId` og language-filer hadde blitt helt flate objekter, som bare beskriver hva det er, og ikke har noe forhold til hvor den skal brukes. Men ganske lagt til å få til noe sånn nå dessverre :(

..og hvis vi hadde hatt det er det mulig vi hadde ønska at det var i en logisk objektstruktur i stedet 😅 #programmerlife